src/app/query/stacks/fees/fees.hooks.ts

```diff
@@ -47,63 +55,65 @@ function useFeeEstimationsMinValues() {
+}: ParseStacksTxFeeEstimationResponseArgs): Fees {
+  if (!!feeEstimation.error) defaultStacksFees;
+  if (txByteLength && feeEstimationQueryFailedSilently(feeEstimation)) {
+    return {
+      blockchain: 'stacks',
+      estimates: getDefaultSimulatedFeeEstimations(txByteLength),
+      calculation: FeeCalculationTypes.DefaultSimulated,
+    };
+  }
 
-export function useStacksFeeEstimations(txByteLength: number | null, txPayload: string) {
-  const feeEstimationsMaxValues = useFeeEstimationsMaxValues();
-  const feeEstimationsMinValues = useFeeEstimationsMinValues();
-  const result = useTransactionFeeEstimation(txByteLength, txPayload);
-  const { data: txFeeEstimation, isError, isLoading } = result;
-
-  return useMemo<FeeEstimations>(() => {
-    const feeEstimations = txFeeEstimation?.estimations;
+  const stacksFeeEstimates: StacksFeeEstimate[] = feeEstimation.estimations.map(estimate => {
+    return {
+      fee: createMoney(estimate.fee, 'STX'),
+      feeRate: estimate.fee_rate,
+    };
+  });
 
-    if (!isLoading && isError) {
-      return {
-        blockchain: 'stacks',
-        estimates: defaultFeeEstimations,
-        calculation: FeeCalculationTypes.Default,
-      };
-    }
-    if (txByteLength && feeEstimationQueryFailedSilently(result)) {
-      return {
-        blockchain: 'stacks',
-        estimates: getDefaultSimulatedFeeEstimations(txByteLength),
-        calculation: FeeCalculationTypes.DefaultSimulated,
-      };
-    }
-    if (feeEstimations?.length) {
-      const feeEstimationsWithCappedValues = getFeeEstimationsWithCappedValues(
-        feeEstimations,
-        feeEstimationsMaxValues,
-        feeEstimationsMinValues
-      );
-      return {
-        blockchain: 'stacks',
-        estimates: feeEstimationsWithCappedValues,
-        calculation: FeeCalculationTypes.FeesCapped,
-      };
-    }
+  if (feeEstimation.estimations && feeEstimation.estimations.length) {
+    const feeEstimationsWithCappedValues = getFeeEstimationsWithCappedValues(
+      stacksFeeEstimates,
+      maxValues,
+      minValues
+    );
     return {
       blockchain: 'stacks',
-      estimates: feeEstimations ?? [],
-      calculation: FeeCalculationTypes.Api,
+      estimates: feeEstimationsWithCappedValues,
+      calculation: FeeCalculationTypes.FeesCapped,
     };
-  }, [
-    feeEstimationsMaxValues,
-    feeEstimationsMinValues,
-    isError,
-    isLoading,
-    result,
-    txByteLength,
-    txFeeEstimation?.estimations,
-  ]);
+  }
+  return {
+    blockchain: 'stacks',
+    estimates: stacksFeeEstimates ?? [],
+    calculation: FeeCalculationTypes.Api,
+  };
+}
+
+export function useStacksFees(txByteLength: number | null, transactionPayload: string) {
+  const feeEstimationsMaxValues = useFeeEstimationsMaxValues();
+  const feeEstimationsMinValues = useFeeEstimationsMinValues();
+
+  return useGetStacksTransactionFeeEstimationQuery(txByteLength, transactionPayload, {
+    onError: () => defaultStacksFees,
```

Comment:
Are you sure this is how it works?

<img width="636" alt="image" src="https://user-images.githubusercontent.com/1618764/206670063-f3ba5135-7717-4189-a6de-4f47e0e13510.png">

I think the callback is fired on error, but it doesn't take the return value and set that as the query response

Reply:
Yeah, I'll remove ...meant to come back to it, thx.